### ISSUE TITLE
Comments on Specs. 

### ISSUE THREAD

**Issue by rvanboening** [rvanboening/Company-Coffee#1]
Nice work. I think it's close, but I found some probably mistakes in the model relationships. Please make an example database (just by hand – not using SQLite or anything) with about 20 rows of "real" data per table. I want to see that example, because I think building it will reveal some of the issues with your model design.

Don't worry about the gem for now. Lots of options I can help you with for that.

LINK TO GOOGLE DOCS. 
https://docs.google.com/presentation/d/13Ia4nFeF2GXdTTzG0LiyfAFHC6c3hiaZ_yaVvH_dMAM/edit?usp=sharing

I rearranged my model design after building mock tables. I don't actually think I need a barista table. Also the order table will simply be a compilation between the menu table and the user table. The Barista will be able to access everything they need from the order and menu table.

**Comment by sumeetjain** [comment]
A table should be named following this rule: "The table's name is the plural of whatever each, individual row represents." So I'd like you to rename "menu" to "items", because each row represents an item that can be ordered. While it's true that all of those items together represent a menu, the convention is that we name tables in the purest, most accurate way possible. Calling it "menu" is adding additional information to the table (i.e. that the table's contents will be viewed as a list of options to choose from). So go with "items".

---

I think you actually want the following relationships:
- `User` / `Order`
  - User has many Orders.
  - Order belongs to a User
- `Order` / `Item`
  - Item has many Orders
  - Order belongs to an Item

I'd like you to make the necessary changes to your spec. You don't need any other tables/models. When you begin your project, I'd like you to stick with jut this much.

I'd also like you to think about why you **shouldn't** have a relationship between `Item` and `User`. In your spec, you relate them. But it's a mistake: Why do you think that is?

---

In your example database, you show columns like 'sugar' and 'decaf' in the "menu" (a.k.a. "items") table. What is their purpose? What information are they storing?

**Comment by rvanboening** [comment]
Making those connections can be hard. I guess in my mind a user will order from items. But really I should be thinking of it as a user has an order and an order is made of items. Oh wow, Just typing that out made me get it. ...1.user will 2.order from 3.the menu/items. √  I shouldn't connect them because I would just be repeating information. 

I have this idea some of the rows in the columns will just be: nil or full, 1 or 0, true or false. That way if it is marked they want it I can print that on my ticket, and if not it will simply be skipped. I was trying to set up the database earlier and wondered if there is an option like that. I was marking them as an INTEGER. And going to write if 1 then sugar and 0 omit. Any ideas?

**Comment by sumeetjain** [comment]
I still don't understand the "sugar"-type of columns in the "items" table. Can you give examples of a "user story"? E.g. explain what the user is going to do and how those columns would be used/affected.

**Comment by rvanboening** [comment]
As a user I want to order my coffee just the way I like it. The menu is more of a check sheet of what my options for ordering are.  
Is this a user story?

**Comment by sumeetjain** [comment]
Yup, more or less. So, given that user story, why are your order's details
being stored in the "items" table instead of the "orders" table?

Run through 3 example user orders where and write down all the rows that
would get added to tables as the result. If you take this exercise
seriously, you should gain a better understanding for how the program will
work.

On Feb 15, 2015, at 9:08 PM, Rachel VanBoening notifications@github.com
wrote:

As a user I want to order my coffee just the way I like it. The menu is
more of a check sheet of what my options for ordering are.

Is this a user story?

**Comment by rvanboening** [comment]
After some thought. I think I only need two tables. I think I just need orders and users. An Order belongs to a user and a User has many orders.

I attached a couple more pictures into my google docs. One of them is a rough work flow. This allowed me to see all of the different options each drink might have. Now I have a full orders table. 

Some of my initial concerns are having null placeholders. How will a null placeholder be printed out. But, I think I can cross that bridge later. 

https://docs.google.com/presentation/d/13Ia4nFeF2GXdTTzG0LiyfAFHC6c3hiaZ_yaVvH_dMAM/edit?usp=sharing

**Comment by sumeetjain** [comment]
Okay, cool. This is a good place to start. :+1: :rocket:
